### ISSUE TITLE
Can't compile with disabled packages

### ISSUE THREAD

**Issue by absolutelynothinghere** [joncampbell123/dosbox-x#3160]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

Void Linux 2021-12-21

### What version(s) of DOSBox-X have this bug?

0.83.20 SDL1

### Describe the bug

I'm trying to build a minimal version of DOSBox-X, by editing the second to last line in the `build-debug` file: `./configure --enable-core-inline--enable-xbrz=no --enable-d3d9=no --enable-d3d-shaders=no --disable-avcodec --enable-force-menu-sdldraw --enable-scaler-full-line --disable-printer --disable-x11 --disable-opengl --disable-freetype --prefix=/usr "$@" $opt || exit 1`

I receive the following error during compilation:

```
parport.cpp: In member function 'virtual void PARALLEL::Run()':
parport.cpp:563:14: error: 'PARALLEL_TYPE_PRINTER' was not declared in this scope; did you mean 'PARALLEL_TYPE_COUNT'?
  563 |    if (mode==PARALLEL_TYPE_PRINTER&&parallelPortObjects[port-1]->parallelType!=PARALLEL_TYPE_PRINTER&&testParallelPortsBaseclass->printer_used) {
      |              ^~~~~~~~~~~~~~~~~~~~~
      |              PARALLEL_TYPE_COUNT
parport.cpp:563:131: error: 'class PARPORTS' has no member named 'printer_used'
  563 | lelType!=PARALLEL_TYPE_PRINTER&&testParallelPortsBaseclass->printer_used) {
      |                                                             ^~~~~~~~~~~~

parport.cpp:578:20: error: 'PARALLEL_TYPE_PRINTER' was not declared in this scope; did you mean 'PARALLEL_TYPE_COUNT'?
  578 |   } else if (mode==PARALLEL_TYPE_PRINTER&&testParallelPortsBaseclass->printer_used) {
      |                    ^~~~~~~~~~~~~~~~~~~~~
      |                    PARALLEL_TYPE_COUNT
parport.cpp:578:71: error: 'class PARPORTS' has no member named 'printer_used'
  578 | f (mode==PARALLEL_TYPE_PRINTER&&testParallelPortsBaseclass->printer_used) {
      |                                                             ^~~~~~~~~~~~
```

### Expected behavior

A successful compilation

### Steps to reproduce the behaviour

1. Edit the `build-debug` file by replacing the second to last line with the following: `./configure --enable-core-inline--enable-xbrz=no --enable-d3d9=no --enable-d3d-shaders=no --disable-avcodec --enable-force-menu-sdldraw --enable-scaler-full-line --disable-printer --disable-x11 --disable-opengl --disable-freetype --prefix=/usr "$@" $opt || exit 1`
2. Run `./build-debug` in the source tree.

**Comment by Wengier** [comment]
Can you build the source code based on the latest code in the repository rather than the source code of version 0.83.20? I don't think the posted error(s) should occur with the latest repository code.

**Comment by grapeli** [comment]
This is the same compile error. It was reported #3117.
It has been [fixed](https://github.com/joncampbell123/dosbox-x/commit/821fa784bb2dcda712c7e9cb592a16fad755e98d).

The linking bug has been [fixed](https://github.com/joncampbell123/dosbox-x/commit/ab7fe1574c19502acbb8e89ed83f81204c09261c) as well.
Use the latest version or backport patches necessary for proper compilation and linking to 0.83.20.

**Comment by absolutelynothinghere** [comment]
Now I get a different error, which I'm guessing is OpenGL related even though I explicitly used `--disable-opengl`... Here it is:

```
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_Shutdown':
/home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:335: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:336: undefined reference to `glFlush'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:344: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:345: undefined reference to `glFlush'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_MakeCurrent':
/home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:367: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:368: undefined reference to `glFlush'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:381: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:382: undefined reference to `glFlush'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/linux-host/lib/libSDL.a(SDL_x11gl.o): in function `X11_GL_CreateContext':
/home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:269: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:270: undefined reference to `glFlush'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:279: undefined reference to `glFinish'
/usr/bin/ld: /home/anon/Downloads/dosbox-x-master/vs/sdl/src/video/x11/SDL_x11gl.c:280: undefined reference to `glFlush'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:469: dosbox-x] Error 1
```

**Comment by grapeli** [comment]
The error is only due to your inexperience and the fact that your build is highly non-standard.
You are building dosbox-x against the supplied [SDL1.x](https://github.com/joncampbell123/dosbox-x/tree/master/vs/sdl) library.

It has opengl enabled by default.
`--enable-video-opengl   include OpenGL context creation [default=yes]`
If you don't need it, you need to modify the [build-dosbox.sh](https://github.com/joncampbell123/dosbox-x/blob/master/vs/sdl/build-dosbox.sh#L26) file with --disable-opengl in the line `26`.
`--disable-video-opengl`

You have to do the same if you want to give up x11.
`--disable-video-x11 --disable-video-opengl`

**Comment by absolutelynothinghere** [comment]
Thank you for the help, I'm closing this since the original issue has been fixed.

Re-enabling OpenGL and X11 compiles without issues. Disabling them produces new errors even after modifying `build-dosbox.sh`, so would it be better to use the SDL2 version in that case? I apologize for my inexperience.

**Comment by grapeli** [comment]
@absolutelynothinghere 
The version with opengl turned off and x11 turned off additionally builds correctly on the basis of the included sdl1.x library.
I checked it before writing the previous post. Somewhere you are making a mistake.

Of course, I recommend using sdl2. If you don't have much experience with software compilation then I would recommend the default configuration. With the use of freetype, opengl.

There was actually a bug building dosbox-x without the freetype library. As I mentioned it **has been fixed**.
